packages/credential-provider-process/src/index.spec.ts

```diff
@@ -0,0 +1,627 @@
+        __addFsMatcher(toMatch: string, toReturn: string): void;
+        __clearFsMatchers(): void;
+        readFile: (path: string, encoding: string, cb: (err: Error|null, data?: string) => void) => void
+    }
+
+    const fs: FsModule = <FsModule>jest.genMockFromModule('fs');
+    let matchers = new Map<string, string>();
+
+    function readFile(
+        path: string,
+        encoding: string,
+        callback: (err: Error|null, data?: string) => void
+    ): void {
+        if (matchers.has(path)) {
+            callback(null, matchers.get(path));
+            return;
+        }
+
+        callback(new Error('ENOENT: no such file or directory'));
+    }
+
+    fs.__addFsMatcher = function(toMatch: string, toReturn: string): void {
+        matchers.set(toMatch, toReturn);
+    };
+    fs.__clearFsMatchers = function(): void {
+        matchers.clear();
+    };
+    fs.readFile = readFile;
+
+    return fs;
+});
+import * as fs from 'fs';
+const {__addFsMatcher, __clearFsMatchers} = fs as any;
+
+jest.mock('os', () => {
+    interface OsModule {
+        homedir: () => string;
+    }
+
+    const os: OsModule = <OsModule>jest.genMockFromModule('os');
+    const path = require('path');
+
+    os.homedir = () => path.sep + path.join('home', 'user');
+
+    return os;
+});
+import {homedir} from 'os';
+import { stringify } from 'querystring';
+
+jest.mock('child_process', () => {
+    interface ChildProcessModule {
+        exec: (command: string, cb: (err: Error|null, stdout?: string) => void) => void
+        __addChildProcessMatcher(toMatch: string, toReturn: string): void;
+        __clearChildProcessMatchers(): void;
+    }
+
+    const child_process = <ChildProcessModule>jest.genMockFromModule('child_process');
+    let matchers = new Map<string, string>();
+
+    function exec(
+        command: string,
+        callback: (err: Error|null, stdout?: string) => void
+    ): void {
+        if (matchers.has(command)) {
+            callback(null, matchers.get(command));
+            return;
+        }
+
+        callback(new Error('ENOENT: no such file or directory'));
+    }
+
+
+    child_process.__addChildProcessMatcher = function(command: string, json: string): void {
+        matchers.set(command, json);
+    };
+    child_process.__clearChildProcessMatchers = function(): void {
+        matchers.clear();
+    };
+
+    child_process.exec = exec;
+
+    return child_process;
+})
+import * as child_process from 'child_process';
+const {__addChildProcessMatcher, __clearChildProcessMatchers} = child_process as any;
+
+const DEFAULT_CREDS = {
+    accessKeyId: 'AKIAIOSFODNN7EXAMPLE',
+    secretAccessKey: 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
+    sessionToken: 'sessionToken',
+};
+
+const FOO_CREDS = {
+    accessKeyId: 'foo',
+    secretAccessKey: 'bar',
+    sessionToken: 'baz',
+};
+
+const DEFAULT_CREDS_JSON = `
+{
+    "AccessKeyId":"AKIAIOSFODNN7EXAMPLE",
+    "SecretAccessKey":"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
```

Comment:
This is a very minor issue but can you use some *fake* credentials strings like 'foo', 'bar'? I know they are not valid but it may make some tools or GitHub believe you really put your real AWS credentials in your codebase